db/rawdb/accessors_chain.go

```diff
@@ -1287,6 +1287,42 @@ func ReadReceiptsCacheV2(tx kv.TemporalTx, block *types.Block, txNumReader rawdb
 	return res, nil
 }
 
+// ReadStateSyncReceiptByHash reads the state-sync receipt by its transaction hash within a specific block.
+// the state-sync transaction (if present) is always the last one in the block.
+// A state-sync receipt is identified if:
+//   - It has CumulativeGasUsed == 0, OR
+//   - Its CumulativeGasUsed equals that of the previous receipt (indicating zero gas usage).
+//
+// If no such receipt is found, nil is returned.
+func ReadStateSyncReceiptByHash(tx kv.TemporalTx, block *types.Block, txNumReader rawdbv3.TxNumsReader) (*types.Receipt, error) {
+	receipts, err := ReadReceiptsCacheV2(tx, block, txNumReader)
```

Comment:
It might be worth confirming during testing if this function returns bor receipts from db or not. Also, I believe the implementation derives some fields based on checking the transaction index in receipt and uses `block.Transactions()` to match. Because state-sync transactions are not part of this list, we might have to handle it separately. 
CC @kamuikatsurgi

Reply:
I checked that you've written a test for it but as you're explicitly writing that to cache, it's expected that this function works correctly. My point was directed for live chains where we're not sure at what point cache is being updated (and whether they contain bor receipts or not).

Reply:
So if `ReadReceiptsCacheV2` returns state-sync transactions, then we don't need this function at all as we'll be able to get all receipts (normal + state-sync) in a single call. If not, yeah we'll have to find a way to fetch state-sync tx from db (not sure in which table or against which key they're stored). 

I just commented so that we ensure it works during testing.